apps/ledger-live-desktop/src/renderer/Default.tsx

```diff
@@ -177,7 +180,38 @@ const NightlyLayer = React.memo(NightlyLayerR);
 
 export default function Default() {
   const history = useHistory();
-  const hasCompletedOnboarding = useSelector(hasCompletedOnboardingSelector);
+  const dispatch = useDispatch();
+  const hasCompletedOnboardingViaLL = useSelector(hasCompletedOnboardingSelector);
+
+  const recoverServices = useFeature("protectServicesDesktop");
+  const recoverStoreId = recoverServices?.params?.protectId ?? "";
+
+  const [recoverOnboardingStatus, setRecoverOnboardingStatus] = useState<boolean>();
+
+  const getRecoverOnboardingStatus = useCallback(() => {
+    const userId = getUserId();
+    const status = getStoreValue(`USER_RESTORE_ONBOARDING_STATUS-${userId}`, recoverStoreId);
+    setRecoverOnboardingStatus(status === "true");
+  }, [recoverStoreId]);
```

Comment:
As this logic is reused on the other component, could you create a custom hook to share the logic, please?